### PR TITLE
gnvim: init at 0.1.5

### PR DESCRIPTION
**Pull request by minijackson**
###### Motivation for this change

[Gnvim](https://github.com/vhakulinen/gnvim) a rich Neovim GUI without any web bloat

Based on the neovim-qt derivation.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by teto — CHANGES_REQUESTED** [review]
very cool that you can arbitrarily set the version (unlike me fetching tags https://github.com/teto/home/blob/master/config/nixpkgs/overlays/pkgs/gnvim/default.nix). 
The wrapper should be in another file so that one can override the unwrapped version.

**Comment by minijackson** [comment]
@teto I think so too, so should we change the neovim-qt derivation to have a separate unwrapped package? Neovim has quite a lot of GUIs, so it would be nice to have a consistent structure.

I didn't know of the `leaveDotGit` parameter, nice to know!

(Also, damn you are fast)

**Comment by teto** [comment]
> (Also, damn you are fast)

damn, I heard that a lot but I thought it was only girls :p ! would be cool to do it for neovim-qt as well but in a separate PR, let's focus :)

**Comment by minijackson** [comment]
Should be good now ^^

**Comment by teto** [comment]
@GrahamcOfBorg build gnvim

**Comment by teto** [comment]
nix-review builds it fine but at runtime I get
`GnvimScroll error: NeovimError(0, "Vim(call):E117: unknown function : gnvim#cursor_tooltip#hide")`
`echo $GNVIM_RUNTIME_PATH` returned nothing, you might need to add it to the wrapper as explained in the README.

**Comment by minijackson** [comment]
Good catch! It also made me realize that the `.desktop` in the wrapper was still pointing to the unwrapped binary. Gnvim works even better for me now, with all the good pretty colors!

File: pkgs/applications/editors/neovim/gnvim/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  installPhase = ''
+    make install PREFIX="${placeholder "out"}"
+  '';
+
+  meta = with stdenv.lib; {
+    description = "GUI for neovim, without any web bloat";
+    homepage = "https://github.com/vhakulinen/gnvim";
+    license     = licenses.mit;
+    maintainers =  with maintainers; [ minijackson ];
+    #inherit (neovim.meta) platforms;
```
Comment (teto) [inline pkgs/applications/editors/neovim/gnvim/default.nix]: remove

**Review by teto — APPROVED** [review]
looks ok apart from the remaining comment.

File: pkgs/applications/editors/neovim/gnvim/default.nix (hunk 1 above)
Comment (minijackson) [reply pkgs/applications/editors/neovim/gnvim/default.nix]: Oh damn I missed that x)

Comment (minijackson) [reply pkgs/applications/editors/neovim/gnvim/default.nix]: Removed!

**Comment by teto** [comment]
@GrahamcOfBorg build gnvim

**Comment by teto** [comment]
Thanks for maintaining this. I wanted in it nixpkgs but didn't have the courage to put myself as a maintainer xD

**Pull request merged**